Corona-Warn-App/src/main/java/de/rki/coronawarnapp/bugreporting/censors/submission/RatQrCodeCensor.kt

```diff
@@ -0,0 +1,62 @@
+package de.rki.coronawarnapp.bugreporting.censors.submission
+
+import dagger.Reusable
+import de.rki.coronawarnapp.bugreporting.censors.BugCensor
+import de.rki.coronawarnapp.bugreporting.censors.BugCensor.Companion.toNewLogLineIfDifferent
+import de.rki.coronawarnapp.bugreporting.censors.BugCensor.Companion.withValidName
+import de.rki.coronawarnapp.bugreporting.debuglog.LogLine
+import de.rki.coronawarnapp.coronatest.qrcode.RapidAntigenHash
+import de.rki.coronawarnapp.util.CWADebug
+import org.joda.time.LocalDate
+import org.joda.time.format.DateTimeFormat
+import javax.inject.Inject
+
+@Reusable
+class RatQrCodeCensor @Inject constructor() : BugCensor {
+
+    override suspend fun checkLog(entry: LogLine): LogLine? {
+
+        val dataToCensor = dataToCensor ?: return null
+
+        var newMessage = entry.message
+
+        with(dataToCensor) {
+            newMessage = newMessage.replace(rawString, "RatQrCode/ScannedRawString")
+
+            newMessage = if (CWADebug.isDeviceForTestersBuild) {
+                newMessage.replace(hash, PLACEHOLDER + hash.takeLast(28))
+            } else {
+                newMessage.replace(hash, PLACEHOLDER + hash.takeLast(4))
+            }
+
+            withValidName(firstName) { firstName ->
+                newMessage = newMessage.replace(firstName, "RATest/FirstName")
+            }
+
+            withValidName(lastName) { lastName ->
+                newMessage = newMessage.replace(lastName, "RATest/LastName")
+            }
+
+            val formatter = DateTimeFormat.forPattern("yyyy-MM-dd")
```

Comment:
Please move this to class level or a companion object, `checkLog` will be called for every line, creating a new  `DateTimeFormat` instance for every logline is costly :money_with_wings: